types/dialogflow/index.d.ts

```diff
@@ -892,6 +905,36 @@ export type MatchMode =
   | "MATCH_MODE_HYBRID"
   | "MATCH_MODE_ML_ONLY";
 
+// API version for the agent.
+export enum ApiVersion {
```

Comment:
Enums from TypeScript won't be compatible with the enums returned in the API. Union types of strings can be used instead where enums are used in the Dialogflow API. These provide type-safety, intellisense, and other advantages of TypeScript.

Reply:
For an example of how gRPC enums are handled with types take a look at the Dialogflow docs for AutoExpansionMode[0] and the corresponding Dialogflow types[1].

0: https://cloud.google.com/dialogflow/docs/reference/rpc/google.cloud.dialogflow.v2#google.cloud.dialogflow.v2.EntityType.AutoExpansionMode
1: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e5b9aa9887930b1879bffe09a1ee50a90f2a2e8a/types/dialogflow/index.d.ts#L960

Reply:
To expand on what tom said, the typings are only meant to reflect what the js already exports. The js does not export an object called ApiVersion with properties API_VERSION_UNSPECIFIED and so on. In methods that expect this "enum", they take strings as arguments and these are simply the possible string values.

eg `export type ApiVersion = "API_VERSION_UNSPECIFIED" | "API_VERSION_V1" ... `

Reply:
Makes sense, thanks for clarifying